heron/packing/src/java/com/twitter/heron/packing/binpacking/FirstFitDecreasingPacking.java

```diff
@@ -151,6 +155,36 @@ public PackingPlan pack() {
     return new PackingPlan(topology.getId(), containerPlans);
   }
 
+  /**
+   * Get a new packing plan given an existing packing plan and component-level changes
+   *
+   * @return new packing plan
+   */
+  public PackingPlan repack(PackingPlan currentPackingPlan, Map<String, Integer> componentChanges) {
+    // Get the instances using FFD allocation
+    Map<Integer, List<String>> ffdAllocation =
+        getFFDAllocation(currentPackingPlan, componentChanges);
+
+    // Construct the PackingPlan
+    Map<String, Long> ramMap = TopologyUtils.getComponentRamMapConfig(topology);
+
+    Set<PackingPlan.ContainerPlan> containerPlans = PackingUtils.buildContainerPlans(
+        ffdAllocation, ramMap, defaultInstanceResources, paddingPercentage);
+
+    //merge the two plans
+    Set<PackingPlan.ContainerPlan> totalContainerPlans = new HashSet<>();
+    totalContainerPlans.addAll(containerPlans);
+    totalContainerPlans.addAll(currentPackingPlan.getContainers());
+
+    LOG.info(String.format("Created a packing plan with %d containers",  containerPlans.size()));
+    for (PackingPlan.ContainerPlan containerPlan : containerPlans) {
+      LOG.info(String.format(
+          "Container %s consists of %s", containerPlan.getId(), containerPlan.getInstances()));
```

Comment:
For a large topology this could be a lot of output on the user console. We could keep it for now, but could you sort the container plans by container id? In #1360 I've added a comparator for unit tests, which we might want to pull into `src`. (Or we implement Comparable for ContainerPlan and all of it's child members)